pkg/kubectl/cmd/logs.go

```diff
@@ -191,22 +194,50 @@ func (o *LogsOptions) Complete(f cmdutil.Factory, out io.Writer, cmd *cobra.Comm
 		builder := resource.NewBuilder(mapper, typer, o.ClientMapper, decoder).
 			NamespaceParam(o.Namespace).DefaultNamespace().
 			SingleResourceType()
-		if o.ResourceArg != "" {
-			builder.ResourceNames("pods", o.ResourceArg)
-		}
-		if selector != "" {
-			builder.ResourceTypes("pods").SelectorParam(selector)
-		}
-		infos, err := builder.Do().Infos()
-		if err != nil {
-			return err
-		}
-		if selector == "" && len(infos) != 1 {
-			return errors.New("expected a resource")
+		if o.ResourceArg != "" && selector == "" {
```

Comment:
sorry, I'm not sure why we should change to `len()`, is that because `len()` is much more efficiency than string compare ?